### PR TITLE
Allow translations of 'yes' in crafting search

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: I18N "Allow translations of 'yes' and 'no' in crafting search"

#### Purpose of change
@Pupsi-Mupsi pointed out in #26452 that the new crafting search dialog suggests that people can search for 'yes' or 'no' in their own language.  Make that be the case.

#### Describe the solution
Check for translations of 'yes' along with the other yes-like strings.

Not required explicitly for 'no' because anything else is treated as 'no'.

#### Describe alternatives you've considered
Could change the example to just be 'y' rather than 'yes' (which also works, and probably wouldn't translate), but it seems better to allow this.

#### Additional context
I tested it in the German translation.  Seems to work as expected.

### CODE REVIEW

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -878,7 +878,8 @@ static bool query_is_yes( const std::string &query )
     const std::string subquery = query.substr( 2 );
 
     return ( ( subquery == "yes" ) || ( subquery == "y" ) || ( subquery == "1" ) ||
-             ( subquery == "true" ) || ( subquery == "t" ) || ( subquery == "on" ) );
+             ( subquery == "true" ) || ( subquery == "t" ) || ( subquery == "on" ) ||
+             ( subquery == _( "yes" ) ) );
```
Comment (BevapDin) [inline src/crafting_gui.cpp]: You can remove nearly all the parentheses in this statement, only the ones around `"yes"` are needed, the others are redundant.

Comment (jbytheway) [reply src/crafting_gui.cpp]: Yes, I thought about that.  I guess maybe it's worth doing while I'm here.

Comment (Pupsi-Mupsi) [reply src/crafting_gui.cpp]: Thanks, for helping out so quickly! Is the "no" also covered with this?

Comment (jbytheway) [reply src/crafting_gui.cpp]: There's no explicit check for 'no'.  Anything that doesn't match one of the 'yes' options is treated as 'no'.

Comment (jbytheway) [reply src/crafting_gui.cpp]: I've updated the OP to clarify that.

**Pull request merged**